### PR TITLE
Update rescript-schema up to V8

### PR DESCRIPTION
**Pull request by DZakh**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,16 +55,15 @@
     "jointz": "7.0.4",
     "json-decoder": "1.4.1",
     "mol_data_all": "1.1.1123",
-    "@mondrian-framework/model": "2.0.43",
```
Comment (DZakh) [inline package.json:58]: It was duplicated

File: cases/rescript-schema.ts [hunk 2]
```diff
@@ -2,8 +2,24 @@ import * as S from 'rescript-schema';
 
 import { createCase } from '../benchmarks';
 
-createCase('rescript-schema', 'parseSafe', () => {
-  const schema = S.object({
+// To bypass the "Assertions require every name in the call target to be declared with an explicit type annotation.ts(2775)"
+// when using assert
+type Data = {
```
Comment (moltar) [inline cases/rescript-schema.ts:7]: Can you please clarify this requirement?

Re-declaring types go against the ideals of this test repository.

Comment (DZakh) [reply cases/rescript-schema.ts]: It's a TS thing. I've managed to work around it by adding `!` after `assert`. The `effect-schema` lib uses the same approach, so probably it's fine. Not super experience with TS hacks :)

**Comment by DZakh** [comment]
@hoeck I'm interested in seeing the performance changes for v8. Could you merge the PR, please? 👀

File: cases/rescript-schema.ts (hunk 2 above)
Comment (hoeck) [reply cases/rescript-schema.ts]: Just FYI: the issue seems to be that you cannot use a generic  `assert arg is T` as a method without explicit type annotations: https://github.com/microsoft/TypeScript/issues/34596

Your hack with the exclamation mark disables the type-assertion, `data` after the assertion is still `unknown`.

Other libraries either do not use an asserts type method at all or they are using a asserts function.

**Comment by hoeck** [comment]
@DZakh thanks for pinging me, the mail from github somehow slipped off my radar :sweat_smile: sorry

**Review by hoeck — APPROVED** [review]

**Comment by DZakh** [comment]
Btw, node v22 runner did break https://github.com/moltar/typescript-runtime-type-benchmarks/actions/runs/10008222868/job/27664468375

**Comment by hoeck** [comment]
> Btw, node v22 runner did break https://github.com/moltar/typescript-runtime-type-benchmarks/actions/runs/10008222868/job/27664468375

Yeah noticed this too, the `npm install` prior to running the linter failed with an NPM internal error. Works on my machine though so let's just run it again and hope for the best.

**Comment by hoeck** [comment]
Ok, this seems to be a bug in the latest NPM version used by nodejs 22: https://github.com/npm/cli/issues/7672

If I am reading this right, waiting for https://github.com/actions/node-versions/pull/182 should give us a fixed node/npm version the next time we run the action (maybe delete the action cache before, don't know).

**Comment by hoeck** [comment]
Now it went through.

**Pull request merged**